addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -1249,6 +1259,24 @@ export class Rtc extends Record {
         this.state.logs["hasTurn"] = hasTurn(this.iceServers);
     }
 
+    async collectLogsFromAllTabs() {
+        this.logSnapshot();
+        return new Promise((resolve) => {
+            const allLogs = { ...this.state.globalLogs };
+            const logCollector = ({ data }) => {
+                if (data.type === CROSS_TAB_CLIENT_MESSAGE.UPDATE_LOGS) {
+                    Object.assign(allLogs, data.logs);
+                }
+            };
+            browser.setTimeout(() => {
+                this._broadcastChannel.removeEventListener("message", logCollector);
+                resolve(allLogs);
+            }, 1000);
+            this._broadcastChannel.addEventListener("message", logCollector);
+            this._postToTabs({ type: CROSS_TAB_CLIENT_MESSAGE.REQUEST_LOGS });
+        });
+    }
+
```

Comment:
I don't understand why the `_broadcastChannel.on/off` are needed here: the start of RTC service in each tab have their broadcast channel listener up and running?

Also note that some tabs could be frozen / throttled. I don't think a late tab will causes issue (except some logs will be missed). Safari (and chromium based browsers in recommended opt-in mode) throttle tabs aggressively, so I wonder if missing some "recent" logs in another tab can be an issue for debugging.

If the logs are sent to shared worker, which covers most browsers, then the logs will live in a non-throttled environment and we won't loose logs. This would be better to send logs to shared worker on update rather than `BroadcastChannel` because of the tab throttling issue.

Reply:
@alexkuhn I wasn't sure about using the worker memory, but I will make a poc